From 8f26d00f309e6d12ac93d7b3dd3aa9da10125335 Mon Sep 17 00:00:00 2001
From: Rudolf Tammekivi <rtammekivi@gmail.com>
Date: Fri, 18 Jan 2013 17:45:25 +0200
Subject: [PATCH 098/100] hwcomposer: Use RGB_BORDERFILL only on MDP 4.2 and
 higher.

MDP 4.0 does not support RGB_BORDERFILL and causes the display to fail
initializing.

Patch Set 2: Actually set MDP_RGB format if not BORDERFILL.

Change-Id: Ib2007221c4dc78fd17ced97c0a0d28896c877b80
---
 libhwcomposer/hwc_mdpcomp.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libhwcomposer/hwc_mdpcomp.cpp b/libhwcomposer/hwc_mdpcomp.cpp
index 4d64fa2..e2513b6 100644
--- a/libhwcomposer/hwc_mdpcomp.cpp
+++ b/libhwcomposer/hwc_mdpcomp.cpp
@@ -193,6 +193,11 @@ bool MDPComp::setupBasePipe(hwc_context_t *ctx) {
     memset(&ovData, 0, sizeof(msmfb_overlay_data));
 
     ovInfo.src.format = MDP_RGB_BORDERFILL;
+    int mdpVersion = qdutils::MDPVersion::getInstance().getMDPVersion();
+    if (mdpVersion >= qdutils::MDP_V4_2)
+        ovInfo.src.format = MDP_RGB_BORDERFILL;
+    else
+        ovInfo.src.format = MDP_RGBA_8888;
     ovInfo.src.width  = fb_width;
     ovInfo.src.height = fb_height;
     ovInfo.src_rect.w = fb_width;
-- 
1.9.5.msysgit.0

